pipelined/bevy_render2/src/render_resource/shader.rs

```diff
@@ -288,13 +392,49 @@ impl ShaderProcessor {
         if scopes.len() != 1 {
             return Err(ProcessShaderError::NotEnoughEndIfs);
         }
-        Ok(final_string)
+
+        let processed_source = Cow::from(final_string);
+
+        match &shader.source {
+            Source::Wgsl(_source) => Ok(ProcessedShader::Wgsl(processed_source)),
+            Source::Glsl(_source, stage) => Ok(ProcessedShader::Glsl(processed_source, *stage)),
+            Source::SpirV(_source) => {
+                unreachable!("SpirV has early return");
+            }
+        }
     }
 }
 
+fn apply_import(
+    import_handles: &HashMap<ShaderImport, Handle<Shader>>,
+    shaders: &HashMap<Handle<Shader>, Shader>,
+    import: &ShaderImport,
+    shader: &Shader,
+    final_string: &mut String,
+) -> Result<(), ProcessShaderError> {
+    let imported_shader = import_handles
+        .get(import)
+        .and_then(|handle| shaders.get(handle))
+        .ok_or_else(|| ProcessShaderError::UnresolvedImport(import.clone()))?;
+    match &shader.source {
+        Source::Wgsl(_) => {
+            if let Source::Wgsl(import_source) = &imported_shader.source {
+                final_string.push_str(import_source);
+            }
+        }
+        Source::Glsl(_, _) => todo!(),
+        Source::SpirV(_) => todo!(),
```

Comment:
Is this meant to be left as `todo!` in this PR?
Naively, it seems straightforward to allow importing glsl into a glsl shader, and return an error for all spirv shaders,

Reply:
Nope! I intended to get back to this but clearly failed :smile: 
Should definitely be straightforward

Reply:
Fixed!